packages/cli/src/commands/info/action.ts

```diff
@@ -150,18 +148,24 @@ function processTargets(targets, packageInfos, config) {
   return packageTasks;
 }
 
+function isTargetNoop(target, packageInfos) {
```

Comment:
I got confused about "noop" here because there ACTUALLY as a runner called noop. This is literally changing the shape of the graph :)

We can do a better job of this by actually feeding the target into the runner's "shouldRun()" fn according to the target type. Part of the logic here in optimizing the graph is something I would like to share with the run action as well. 

We can call this entire "info" feature unstable until we figure this out! So I'm inclined to approve on this change, knowing that the "UI" (the CLI command) won't change, but the behavior will change later to accommodate more runner types.

Reply:
Yes, I was bit confused as well. The info command already had a sort of hardcoded check to see if the task was defined as a package.json script, so I just continued with that.

I got the `noop` name from the [task code](https://github.com/microsoft/lage/blob/b366b8d0b7d9319a90cacbc18f2773aa9f0b1e17/packages/target-graph/src/TargetFactory.ts#L26C1-L36C4) but I see now that this property is available in `task.type`, so I think we can use that here by checking for `type === "npmScript"` instead of duplicating the logic.

But I don't know if lage info is only targetting BuildXL, and if BuildXL can only support npm script tasks then we have to ignore other tasks in addition to the "noop" ones.